CMakeLists.txt

```diff
@@ -28,26 +33,42 @@ if(APPLE)
   # Special compile flags for Objective-C++
   set(OBJCXX_COMPILE_FLAGS
     "-x objective-c++ -fobjc-arc -Wno-c++11-extensions -mmacosx-version-min=${APPLE_MIN_ABI}")
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+  set(FREEBSD TRUE)
+  set(CXX_COMPILE_FLAGS "${CXX_COMPILE_FLAGS} -std=c++11 -stdlib=libc++")
+  set(OS_WHOLELINK_PRE "")
+  set(OS_WHOLELINK_POST "")
 else()
-  if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
-    set(FREEBSD TRUE)
-    set(CXX_COMPILE_FLAGS "${CXX_COMPILE_FLAGS} -std=c++11 -stdlib=libc++")
-    set(OS_WHOLELINK_PRE "")
-    set(OS_WHOLELINK_POST "")
-  else()
-    set(LINUX TRUE)
-    set(CXX_COMPILE_FLAGS "${CXX_COMPILE_FLAGS} -std=c++11")
-    set(OS_WHOLELINK_PRE "-Wl,-whole-archive")
-    set(OS_WHOLELINK_POST "-Wl,-no-whole-archive")
-  endif()
+  set(LINUX TRUE)
+  set(CXX_COMPILE_FLAGS "${CXX_COMPILE_FLAGS} -std=c++11")
+  set(OS_WHOLELINK_PRE "-Wl,-whole-archive")
+  set(OS_WHOLELINK_POST "-Wl,-no-whole-archive")
 endif()
 
+# Use osquery language to set platform/os
+execute_process(
+  COMMAND "${CMAKE_SOURCE_DIR}/tools/provision.sh" get_platform
+  WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+  OUTPUT_VARIABLE PLATFORM
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+string(REPLACE "." "_" PLATFORM "${PLATFORM}")
+string(TOUPPER "${PLATFORM}" PLATFORM)
+list(GET PLATFORM 0 OSQUERY_BUILD_OS)
+list(GET PLATFORM 1 OSQUERY_BUILD_DISTRO)
+
 # make debug (environment variable from Makefile)
 if(DEFINED ENV{DEBUG})
   set(CMAKE_BUILD_TYPE "Debug")
-  set(C_COMPILE_FLAGS "${CMAKE_C_FLAGS} -g -DDEBUG -O0 -pg")
+  set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -g -DDEBUG -O0 -pg")
+elseif(DEFINED ENV{SANITIZE})
+  # make santifize (cannot make debug sanitize)
+  set(C_COMPILE_FLAGS "-g -O0 -fno-omit-frame-pointer")
+  set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -fsanitize=leak -fsanitize=address")
 else()
-  set(C_COMPILE_FLAGS "${CMAKE_C_FLAGS} -O2")
+  set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -O2")
+  # Do not enable fortify with clang: http://llvm.org/bugs/show_bug.cgi?id=16821
+  #set(C_COMPILE_FLAGS "${C_COMPILE_FLAGS} -D_FORTIFY_SOURCE=2")
```

Comment:
Unfortunate, but on CentOS7 this is optimizing out parts of code and causing infinite jumps.